broker/broker_test.go

```diff
@@ -0,0 +1,174 @@
+package broker_test
+
+import (
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/pivotal-cf/brokerapi"
+
+	"github.com/18F/cf-cdn-service-broker/broker"
+	"github.com/18F/cf-cdn-service-broker/models"
+)
+
+type FakeRouteManager struct {
+	RouteCreate models.Route
+	ErrorCreate error
+	RouteGet    models.Route
+	ErrorGet    error
+}
+
+func (m *FakeRouteManager) Create(instanceId, domain, origin string) (models.Route, error) {
+	return m.RouteCreate, m.ErrorCreate
+}
+
+func (m *FakeRouteManager) Get(instanceId string) (models.Route, error) {
+	return m.RouteGet, m.ErrorGet
+}
+
+func (m *FakeRouteManager) Update(models.Route) error {
+	return nil
+}
+
+func (m *FakeRouteManager) Disable(models.Route) error {
+	return nil
+}
+
+func (m *FakeRouteManager) Renew(models.Route) error {
+	return nil
+}
+
+func (m *FakeRouteManager) RenewAll() {
+	return
+}
+
+func TestProvisioning(t *testing.T) {
+	suite.Run(t, new(ProvisionSuite))
+}
+
+type ProvisionSuite struct {
+	suite.Suite
+	Broker broker.CdnServiceBroker
+}
+
+func (s *ProvisionSuite) SetupTest() {
+	s.Broker = broker.CdnServiceBroker{
+		Manager: &FakeRouteManager{
+			RouteCreate: models.Route{State: "provisioned"},
```

Comment:
Agreed--this is what I did here originally, but combining custom types with `gorm` is a little annoying. `gorm` doesn't know how to marshal and unmarshal custom types, so our `State` types has to implement the `Valuer` and `Scanner` interfaces. Which seemed like a lot of boilerplate for an enum at the time, but yeah, it's probably worth it.